src/System.Private.CoreLib/src/System/Text/Utf8Span.Manipulation.cs

```diff
@@ -0,0 +1,532 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text.Unicode;
+
+namespace System.Text
+{
+    public readonly ref partial struct Utf8Span
+    {
+        [StackTraceHidden]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void CheckSplitOptions(Utf8StringSplitOptions options)
+        {
+            if ((uint)options > (uint)(Utf8StringSplitOptions.RemoveEmptyEntries | Utf8StringSplitOptions.TrimEntries))
+            {
+                CheckSplitOptions_Throw(options);
+            }
+        }
+
+        [StackTraceHidden]
+        private static void CheckSplitOptions_Throw(Utf8StringSplitOptions options)
+        {
+            throw new ArgumentOutOfRangeException(
+                paramName: nameof(options),
+                message: SR.Format(SR.Arg_EnumIllegalVal, (int)options));
+        }
+
+        public SplitResult Split(char separator, Utf8StringSplitOptions options = Utf8StringSplitOptions.None)
+        {
+            if (!Rune.TryCreate(separator, out Rune rune))
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.separator, ExceptionResource.ArgumentOutOfRange_Utf16SurrogatesDisallowed);
+            }
+
+            CheckSplitOptions(options);
+
+            return new SplitResult(this, rune, options);
+        }
+
+        public SplitResult Split(Rune separator, Utf8StringSplitOptions options = Utf8StringSplitOptions.None)
+        {
+            CheckSplitOptions(options);
+
+            return new SplitResult(this, separator, options);
+        }
+
+        public SplitResult Split(Utf8Span separator, Utf8StringSplitOptions options = Utf8StringSplitOptions.None)
+        {
+            if (separator.IsEmpty)
+            {
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_CannotBeEmptySpan, ExceptionArgument.separator);
```

Comment:
These methods are neither generic nor super perf critical. ThrowHelper is not worth it for them.